components/application-tag-mgt/org.wso2.carbon.identity.application.tag.mgt/src/main/java/org/wso2/carbon/identity/application/tag/mgt/dao/impl/ApplicationTagDAOImpl.java

```diff
@@ -0,0 +1,256 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.tag.mgt.dao.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.tag.common.model.ApplicationTagPOST;
+import org.wso2.carbon.identity.application.tag.common.model.ApplicationTagsListItem;
+import org.wso2.carbon.identity.application.tag.mgt.ApplicationTagMgtException;
+import org.wso2.carbon.identity.application.tag.mgt.constant.ApplicationTagManagementConstants;
+import org.wso2.carbon.identity.application.tag.mgt.constant.SQLConstants;
+import org.wso2.carbon.identity.application.tag.mgt.dao.ApplicationTagDAO;
+import org.wso2.carbon.identity.application.tag.mgt.util.ApplicationTagManagementUtil;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * This class implements the {@link ApplicationTagDAO} interface.
+ */
+public class ApplicationTagDAOImpl implements ApplicationTagDAO {
+
+    private static final Log log = LogFactory.getLog(ApplicationTagDAOImpl.class);
+
+    @Override
+    public String createApplicationTag(ApplicationTagPOST applicationTagDTO, String tenantDomain)
+            throws ApplicationTagMgtException {
+
+        Integer tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        if (log.isDebugEnabled()) {
+            log.debug("Creating Application Tag of " + applicationTagDTO.getName());
+        }
+        String generatedAppTagId = UUID.randomUUID().toString();
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try {
+                PreparedStatement prepStmt = dbConnection.prepareStatement(SQLConstants.ADD_APPLICATION_TAG);
+                prepStmt.setString(1, generatedAppTagId);
+                prepStmt.setString(2, applicationTagDTO.getName());
+                prepStmt.setString(3, applicationTagDTO.getColour());
+                prepStmt.setInt(4, tenantId);
+                prepStmt.execute();
+
+                ResultSet results = prepStmt.getGeneratedKeys();
+
+                String applicationTagId = null;
+                if (results.next()) {
+                    applicationTagId = results.getString(1);
+                }
+                // some JDBC Drivers returns this in the result, some don't
+                if (applicationTagId == null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("JDBC Driver did not return the tag id, executing Select operation");
+                    }
+                    applicationTagId = getApplicationTagByName(applicationTagDTO.getName(), tenantId, dbConnection);
```

Comment:
According to the method comment, you are returning the tag id. But this is not the tag id. For any external purposes, id should be the UUID.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/4904/commits/3d8ec5b161358637ba2c96ed4155edb1ccd619dc